src/main/java/Backend/ExpressionCreator.java

```diff
@@ -144,25 +155,38 @@ private boolean containsOuterBrackets(List<String> terms){
         return true;
     }
 
-    private Expression createExpressionRecursiveHelper(String expressionType, Map<String, Integer> operatorAndIndices,
+    /**
+     * @param expressionType The type of Expression that needs to be created. Can only be 'Operator', 'Comparator'
```

Comment:
This just occurred to me, but could we use enums to enforce this? I am not too familiar with them, but I believe this seems like the kind of situation for them